### PR TITLE
chore(release.yml): Update asset builder go version to v1.21

### PR DESCRIPTION
**Pull request by frauniki**
- issue: https://github.com/FeLvi-zzz/tentez/actions/runs/7458657645/job/20293052337

- Fixed problem with asset build failure
  - Update asset builder go version to v1.21

### CODE REVIEW

**Comment by frauniki** [comment]
v0.10.3 will be re-released

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -37,3 +37,16 @@ jobs:
...
+
+  asset-test-build:
+    name: test build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.mod"
+          go-version: "1.21"
```
Comment (FeLvi-zzz) [inline .github/workflows/test.yaml:49]: > If both the go-version and the go-version-file inputs are provided then the go-version input is used.
https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

you can remove `go-version`.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v2
         with:
-          go-version: 1.18.x
+          go-version: "1.21"
```
Comment (FeLvi-zzz) [inline .github/workflows/release.yml:14]: could you make it same as `.github/workflows/test.yaml`?
```yaml
      - uses: actions/checkout@v4
      - uses: actions/setup-go@v5
        with:
          go-version-file: "go.mod"
```

File: .github/workflows/test.yaml (hunk 1 above)
Comment (frauniki) [reply .github/workflows/test.yaml]: fixed

File: .github/workflows/release.yml (hunk 2 above)
Comment (frauniki) [reply .github/workflows/release.yml]: fixed

**Comment by frauniki** [comment]
@FeLvi-zzz 
Could you review it again?

**Review by FeLvi-zzz — APPROVED** [review]
LGTM

**Pull request merged**